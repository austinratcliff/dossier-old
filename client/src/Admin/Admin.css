.Admin {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: white;
  color: #111;
  font-family: 'IBM Plex Mono', monospace;
  height: 100vh;
  width: 100vw;
}

.Admin .header {
  position: absolute;
  font-size: calc(10px + 5vmin);
  margin: 0;
  animation: fade-out 1s 1s 1 forwards;
}

.Admin .new-project-form {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  opacity: 0;
  visibility: hidden;
  animation: fade-in 1s 2s 1 forwards;
}

.Admin .form-header {
  font-size: calc(10px + 2vmin);
  margin: 0 0 25px 0;
}

.Admin .highlight {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
}

.form-control {
  border: 1px solid #92A8D1 !important;
  border-radius: 0 !important;
  color: #111 !important;
}

.form-text a {
  color: #92A8D1 !important;
  font-weight: bold !important;
  text-decoration: none !important;
}

.btn {
  border: 1px solid #92A8D1 !important;
  border-radius: 0 !important;
  background-color: white !important;
  color: #92A8D1 !important;
  font-weight: bold !important;
  width: 100% !important;
}

.btn:hover {
  background-color: #92A8D1 !important;
  color: white !important;
}

@keyframes fade-out {
  from {opacity: 1; visibility: visible;}
  to {opacity: 0; visibility: hidden;}
}

@keyframes fade-in {
  from {opacity: 0; visibility: hidden;}
  to {opacity: 1; visibility: visible;}
}
