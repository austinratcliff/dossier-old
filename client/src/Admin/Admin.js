import React from 'react'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import './Admin.css'

const Admin = () => {
  return (
    <div className="Admin">
      <h1 className="header">
        <i
          className="fas fa-user-lock"
          title="about"
        /> Admin
      </h1>
      <div className="new-project-form">
        <h2 className="form-header">
          <span className="highlight">
            New project
          </span>
        </h2>
        <Form
          action="/api/projects"
          method="POST">
          <Form.Group>
            <Form.Control
              type="text"
              name="name"
              placeholder="Name"
            />
          </Form.Group>
          <Form.Group>
            <Form.Control
              as="textarea"
              rows="3"
              name="description"
              placeholder="Description"
            />
            <Form.Text>
              <a href="https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet/">
                Markdown cheatsheet
              </a>
            </Form.Text>
          </Form.Group>
          <Form.Group>
            <Form.Control
              type="text"
              name="projectURL"
              placeholder="Project URL"
            />
            <Form.Text>
              https://project.com
            </Form.Text>
          </Form.Group>
          <Form.Group>
            <Form.Control
              type="text"
              name="githubURL"
              placeholder="GitHub URL"
            />
            <Form.Text>
              https://github.com/username/project
            </Form.Text>
          </Form.Group>
          <Form.Group>
            <Form.Check
              inline
              type="checkbox"
              label="Creator"
              name="badges[]"
              value="CREATOR"
            />
            <Form.Check
              inline
              type="checkbox"
              label="Contributor"
              name="badges[]"
              value="CONTRIBUTOR"
            />
            <Form.Check
              inline
              type="checkbox"
              label="Active"
              name="badges[]"
              value="ACTIVE"
            />
            <Form.Check
              inline
              type="checkbox"
              label="Inactive"
              name="badges[]"
              value="INACTIVE"
            />
          </Form.Group>
          <Button type="submit">
            Add
          </Button>
        </Form>
      </div>
    </div>
  )
}

export default Admin
