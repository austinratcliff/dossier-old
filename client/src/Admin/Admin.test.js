import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import Admin from './Admin'

describe('Admin component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Admin />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<Admin />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
