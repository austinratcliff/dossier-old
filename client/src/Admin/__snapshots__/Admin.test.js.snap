// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin component matches snapshot 1`] = `
<div
  className="Admin"
/>
`;
